Color literals count as single character towards line count

VFUC opened the pull request:
Hey! Trying to work on issue #742 - I've added a check for an expanded color literal substring when counting line characters. If found, it subtracts the length of the substring (-1) from the total line length, so that a color literal only counts as a single character towards the line count rule. Feedback welcome 🙂

VFUC commented:
Thanks for the feedback, I've added the requested changes. I have also modified the general approach to support multiple color literals in a single line.

norio-nomura commented on Source/SwiftLintFramework/Rules/LineLengthRule.swift:
```diff
@@ -34,24 +36,24 @@ public struct LineLengthRule: ConfigurationProviderRule, SourceKitFreeRule {
                     options: .LiteralSearch,
-                    range: colorLiteralRangeStart.startIndex..<line.content.endIndex,
+                    range: rangeStart.startIndex..<string.endIndex,
                     locale: nil) {
```
Could you please add indent to lines between `let rangeEnd…` and `locale:nil) {`?

norio-nomura commented on CHANGELOG.md:
```diff
@@ -17,6 +17,10 @@
   [Norio Nomura](https://github.com/norio-nomura)
   [Syo Ikeda](https://github.com/ikesyo)
 
+* Color literals count as single characters to avoid unintentional line length violations.
```
Could you please add 2 trailing spaces to this line for making Markdown adds line ending?

marcelofabri commented:
Maybe this should be improved to handle other literals (such as image ones)?

VFUC commented:
I have now extracted the literal stripping into a function and applied it for color and image literals (as suggested by @marcelofabri , thanks for that!). If  future literals keep the same `#...literal()` structure, adding them will be easy.

I've also added the requested formatting.

norio-nomura commented:
Thanks for doing this! 🙏
